src/resultAsync.ts

```diff
@@ -558,6 +559,32 @@ export class ResultAsync<TValue = Unit, TError = string> {
     );
   }
 
+  /**
+   * Executes the given async action if the ResultAsync is successful or failed
+   * @param action an async function
+   * @returns the current ResultAsync wrapping the inner Result
+   */
+  tapEither(asyncAction: AsyncAction): ResultAsync<TValue, TError>;
+  /**
+   * Executes the given action if the ResultAsync is successful or failed
+   * @param action a function
+   * @returns the current ResultAsync wrapping the inner Result
+   */
+  tapEither(action: Action): ResultAsync<TValue, TError>;
+  tapEither(action: Action | AsyncAction): ResultAsync<TValue, TError> {
+    return new ResultAsync(
```

Comment:
What is the difference in using `new ResultAsync` compared to `ResultAsync.from`?
To me, it looks similar.

Reply:
You're right - they are very similar.

I use the constructor in the instance methods to reduce the call stack by 1 frame and keep things consistent.
We can't use the constructor outside the class because it is marked as `protected`, so we have to use the static methods there.

We still have some confusing scenarios [with ResultAsync.from()](https://github.com/seangwright/typescript-functional-extensions/issues/16#issuecomment-1256850185) that I'd like to try and tackle at some point. By using the constructor internally, we can keep the behavior more predictable than `.from()`.

Reply:
Aah, interesting!

Thanks for the explanation